internal/expr/parser_helper_test.go

```diff
@@ -132,3 +132,47 @@ func (s *ExprInternalSuite) TestUnfinishedQuote(c *C) {
 		}
 	}
 }
+
+func (s *ExprInternalSuite) TestValidComments(c *C) {
+	var p = NewParser()
+
+	validComments := []string{
+		`-- Simple comment`,
+		`-- Simple with line break
+		`,
+		`/* muti
+		 line */`,
+		`/* unfinished mutiline`,
+		`/* -- */`,
+		`-- */`,
+	}
+
+	for _, q := range validComments {
+		p.init(q)
+		ok := p.skipComment()
+		if !ok {
+			c.Errorf("test failed. %s is a valid comment", q)
+		}
+	}
+}
+
+func (s *ExprInternalSuite) TestInvalidComments(c *C) {
+	var p = NewParser()
+
+	invalidComments := []string{
+		`- not comment`,
+		`- - not comment`,
+		`/ * not comment */`,
```

Comment:
Maybe add also: 
\``/ not comment */`\`
\``/* not comment /`\`

Reply:
I think the second one is actually considered a valid comment, first makes sense though

Reply:
Added the first one.